cfme/tests/infrastructure/test_host.py

```diff
@@ -312,3 +312,40 @@ def _finalize():
             actual_state, power_state
         ),
     )
+
+
+@test_requirements.infra_hosts
+def test_infrastructure_hosts_refresh_multi(appliance, setup_provider, provider):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/6h
+        testSteps:
+            1. Navigate to the Compute > Infrastructure > Providers view.
+            2. Click on a provider guadicon, and then the hosts link along the top row of the view.
+            3. Select all hosts (need at least 2 hosts) by checking the box in upper left of
+               quadicons.
+            4. Click "Refresh Relationships and Power States" under the Configuration
+               dropdowm, and then click "OK" when prompted.
+        expectedResults:
+            1. Providers view is displayed.
+            2. Hosts view is displayed.
+            3.
+            4. "Refresh initiated for X Hosts from the CFME Database" is displayed in green
+               banner where "X" is the number of selected hosts. Properties for each host are
+               refreshed.
+    """
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    num_hosts = len(hosts_view.entities.get_all())
+    if num_hosts < 2:
```

Comment:
Just a thought, but not something to implement in this PR.

It would be great if ProviderFilter allowed for specifying a number of fields for required_fields. Right now it just supports looking for the presence of a field, and a matching value if supplied as a tuple.

For this PR, I think it would be better to use a fixture to get the number of hosts on the provider, skipping where there is only one. That way the skip happens in Setup, and not during Call phase.

The collection can actually be accessed directly via `provider.hosts`, an attribute is created by the BaseEntity class.

Reply:
There's actually a subtle assertion going on here I guess, and that's the fact that the UI is displaying 2 or more hosts - its possible that the provider has more than two hosts in the REST collection used for `HostsCollection.all`, but there is  only one displayed on the UI.